data/data-pipeline/data_pipeline/content/config/generate_data_dictionary.py

```diff
@@ -0,0 +1,185 @@
+    yaml_contents = yaml.safe_load(
+        open(yaml_path, "r", encoding="utf-8").read()
+    )
+    if "sheets" in yaml_contents:
+        fields = []
+        for sheet in yaml_contents["sheets"]:
+            # excel yaml has ability for many sheets
+            fields.extend([single_field for single_field in sheet["fields"]])
+    else:
+        # other files do not
+        fields = yaml_contents["fields"]
+
+    to_frame_dict = {CEJST_SCORE_COLUMN_NAME: []} | {
+        field: [] for field, _ in fields_to_store
+    }
+
+    return to_frame_dict, fields
+
+
+def _create_df_from_yaml(
+    yaml_path: pathlib.Path,
+    fields_to_store: list,
+    native_score_column: str = CEJST_SCORE_COLUMN_NAME,
+):
+    """helper function to create a dataframe from yaml"""
+    to_frame_dict, fields = _set_up_infrastructure(
+        yaml_path=yaml_path,
+        fields_to_store=fields_to_store,
+    )
+    for details in fields:
+        to_frame_dict[CEJST_SCORE_COLUMN_NAME].append(
+            details[native_score_column]
+        )
+        for column_name, dict_field in fields_to_store:
+            try:
+                to_frame_dict[column_name].append(details[dict_field])
+            except KeyError:
+                assert column_name != CEJST_SCORE_COLUMN_NAME
+                to_frame_dict[column_name].append(np.nan)
+    return pd.DataFrame(to_frame_dict).set_index(CEJST_SCORE_COLUMN_NAME)
+
+
+def _get_datatype(
+    input_name: str,
+    input_type: str,
+    percentile_string: str = PERCENTILE_STRING,
+    loss_rate_string: str = LOSS_RATE_STRING,
+):
+    """helper to convert datatype"""
+    if percentile_string in input_name:
+        return "percentile"
+    elif loss_rate_string in input_name:
+        return "rate"
+    else:
+        return input_type
+
+
+def _get_calculation_notes(column_name):
+    """produces calculation notes"""
+    calculation_notes = []
+    if PERCENTILE_STRING in column_name:
+        calculation_notes += [PERCENTILE_EXPLANATION]
+    if LOW_STRING in column_name:
+        calculation_notes += [LOW_PERCENTILE_EXPLANATION]
+    if ISLAND_STRING in column_name:
+        calculation_notes += [ISLAND_AREAS_EXPLANATION]
+    return " ".join(calculation_notes)
+
+
+if __name__ == "__main__":
+    # read component yamls
+    csv_codes_df = _create_df_from_yaml(
+        yaml_path=CSV_CONFIG,
+        fields_to_store=[(CSV_LABEL_FIELD, "label"), (CSV_FORMAT, "format")],
+        native_score_column=CEJST_SCORE_COLUMN_NAME,
+    )
+
+    excel_codes_df = _create_df_from_yaml(
+        yaml_path=EXCEL_CONFIG,
+        fields_to_store=[(EXCEL_LABEL_FIELD, "label")],
+        native_score_column=CEJST_SCORE_COLUMN_NAME,
+    )
+
+    score_detail_notes_df = _create_df_from_yaml(
+        yaml_path=SCORE_FIELD_CONFIG,
+        fields_to_store=[
+            (NOTES_FIELD, "notes"),
+            (THRESHOLD_CATEGORY_FIELD, "category"),
+        ],
+        native_score_column="column_name",
```

Comment:
I gotta be honest, I'm scratching my head a bit here about what all of this is doing. Could you add some comments making it a bit more obvious why this is `"column_name"`?

Reply:
Yes! I can also just change it in the yaml.